app/model/message/message.go

```diff
@@ -74,10 +93,26 @@ func NewSignature(sourceChainId, targetChainId uint64, transferID, asset, receiv
 		Amount:        amount,
 		Signature:     signature,
 	}
-	return &Message{topicMsg}
+	return &Message{TopicMessage: &model.TopicMessage{Message: &model.TopicMessage_FungibleSignatureMessage{FungibleSignatureMessage: topicMsg}}}
+}
+
+// NewNftSignature instantiates Signature Message struct ready for submission to the Bridge Topic
+func NewNftSignature(sourceChainId, targetChainId uint64, transferID, asset string, tokenId uint64, metadata, recipient, signature string) *Message {
```

Comment:
You can just accept `model.TopicEthNftSignatureMessage` here as this way the signature will expect just 1 object instead of each field separate and if something changes in the struct's fields it won't need to update the parameters.